vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/validation/ComboBoxBasicValidationIT.java

```diff
@@ -31,38 +32,43 @@ public void required_triggerBlur_assertValidity() {
         $("button").id(REQUIRED_BUTTON).click();
 
         testField.sendKeys(Keys.TAB);
-        assertServerInvalid();
-        assertClientInvalid();
+        assertValidationCount(0);
+        assertServerValid();
+        assertClientValid();
     }
 
     @Test
     public void required_changeValue_assertValidity() {
         $("button").id(REQUIRED_BUTTON).click();
 
         testField.selectByText("foo");
+        assertValidationCount(1);
         assertServerValid();
         assertClientValid();
 
         testField.clear();
+        assertValidationCount(1);
         assertServerInvalid();
         assertClientInvalid();
+    }
+
+    @Test
+    public void required_enterCustomValue_assertValidity() {
+        $("button").id(REQUIRED_BUTTON).click();
 
-        // Try enter custom value
         testField.sendKeys("custom", Keys.TAB);
```

Comment:
**note:** There is no validation because the `combo-box` web component reverts invalid values when custom values are disabled.